pkg/asset/installconfig/powervs/metadata.go

```diff
@@ -300,11 +300,26 @@ func (m *Metadata) GetVPCSubnets(ctx context.Context, vpcName string) ([]vpcv1.S
 
 // GetDNSServerIP gets the IP of a custom resolver for DNS use.
 func (m *Metadata) GetDNSServerIP(ctx context.Context, vpcName string) (string, error) {
+	if m.client == nil {
+		client, err := NewClient()
+		if err != nil {
+			return "", err
+		}
+
+		m.client = client
+	}
```

Comment:
Why can't you initialize the client when you create the metadata via `NewMetadata`?

Reply:
What to do if `NewClient()` returnis `err`? Should `NewMetadata()` start returning `(*Metadata, error)`?

Reply:
Either that or you can receive the client as a parameter to `NewMetadata` so it's already a valid client.

Reply:
I got `NewMetadata()` returning `(*Metadata,error)`, dropped the initialize-as-needed logic in `client.go` functions, and still a private cluster deploying fine on PowerVS in `osa21`. I think the latest force-push is close to what we want. PTAL.